"Black" all the things!

willdurand opened the pull request:
Fixes https://github.com/mozilla/addons-server/issues/8617
Fixes https://github.com/mozilla/addons-server/issues/9885

---

This is going to be fun!

willdurand commented:
I excluded the migrations and re-applied black so that it does not conflict. This is ready to go 😈

willdurand commented:
mm I am confused. It says `toml` is not found but it is installed via `requirements/travis_base.txt`

eviljeff commented on pyproject.toml:
```diff
@@ -0,0 +1,21 @@
+   | \.npm
+   | \.tox
+  )/
+  | src/olympia/wsgi\.py
```
😕 mmm?

willdurand replied:
this was in Chris' patch. I meant to check why but I forgot obviously.

eviljeff commented on src/olympia/access/acl.py:
```diff
@@ -63,41 +64,50 @@ def langpack_submission_allowed(user, parsed_addon_data):
     """
-    return (
-        not parsed_addon_data.get('type') == amo.ADDON_LPAPP or
-        action_allowed_user(user, amo.permissions.LANGPACK_SUBMIT))
+    return not parsed_addon_data.get("type") == amo.ADDON_LPAPP or action_allowed_user(
+        user, amo.permissions.LANGPACK_SUBMIT
+    )
```
this doesn't seem like an improvement, but I'm not going to fight Black on it.

eviljeff commented on Makefile-docker:
```diff
@@ -192,6 +192,7 @@ lint: ## lint the code
 	flake8 src/ services/ tests/
 	$(shell npm $(NPM_ARGS) bin)/prettier --check '**'
 	curlylint src/
+	black src/ services/ tests/
```
this needs to be a command that would return the problems with the code, rather than auto-fix them.  I'd also be happier if it was at the top, before flake8.

willdurand replied:
ah oops, right. there was `--check` before

willdurand replied:
ah oops, right. there was `--check` before

eviljeff commented on Makefile-docker:
```diff
@@ -255,6 +256,7 @@ watch_js_tests: ## Run+watch the JavaScript test suite (requires compiled/compre
 .PHONY: format
 format: ## Autoformat our codebase.
 	$(shell npm $(NPM_ARGS) bin)/prettier --write '**'
+	black src/
```
should we format `/services` too?

willdurand replied:
yep, sorry about that

eviljeff reviewed (commented):
I _do_ wonder how much of this is `'` to `"` (so how much smaller if would be if string normalization was skipped) 🤷‍♂️

willdurand commented:
Green!

willdurand commented:
> I _do_ wonder how much of this is `'` to `"` (so how much smaller if would be if string normalization was skipped) 🤷‍♂️

This would be smaller but we don't have a way to enforce the use of single quotes everywhere, which is why this patch uses black defaults.

willdurand commented:
FTR with double quotes:

<img width="196" alt="Screenshot 2020-12-09 at 16 07 27" src="https://user-images.githubusercontent.com/217628/101647172-b558b000-3a38-11eb-974d-2bd838c577d0.png">

eviljeff commented on src/olympia/scanners/admin.py:
```diff
@@ -48,41 +51,38 @@ def _is_safe_url(url, request):
-                'display': title,
+                "selected": self.value() == lookup,
+                "query_string": cl.get_query_string({self.parameter_name: lookup}, []),
+                "display": title,
```
the "switch back to single quotes " commit should have reverted these changes?  (I just picked a few files to check - I don't know how widespread it is)

willdurand replied:
Yes, I am working on fixing that.

willdurand replied:
I am going to re-apply black entirely to fix conflicts so this shuld be fixed.

willdurand commented:
So I changed the black config as we discussed on Element: black skips string normalization (for now). I'll file an issue about that later. I re-applied black on our codebase. I added another commit to fix errors reported by flake8. I re-ran `black --check` + `flake8` and everything was ok.

I'll take care of the existing double quotes in a follow-up patch.

diox reviewed (approved):
:black_heart:

eviljeff commented on services/utils.py:
```diff
@@ -55,15 +55,18 @@ def user_media_url(what):
 
-version_re = re.compile(r"""(?P<major>\d+)         # major (x in x.y)
+version_re = re.compile(
+    r"""(?P<major>\d+)         # major (x in x.y)
                             \.(?P<minor1>\d+)      # minor1 (y in x.y)
                             \.?(?P<minor2>\d+|\*)? # minor2 (z in x.y.z)
                             \.?(?P<minor3>\d+|\*)? # minor3 (w in x.y.z.w)
                             (?P<alpha>[a|b]?)      # alpha/beta
                             (?P<alpha_ver>\d*)     # alpha/beta version
                             (?P<pre>pre)?          # pre release
                             (?P<pre_ver>\d)?       # pre release version
-                          """, re.VERBOSE)
+                          """,
```
nit: the formatting for this is weird now (not that it matters after it's compiled)

eviljeff commented on src/olympia/abuse/models.py:
```diff
@@ -50,8 +50,11 @@ class AbuseReport(ModelBase):
+        (
+            'SETTINGS',
+            3,
+            'Changes search / homepage / new tab page ' 'without informing user',
```
nit: weird string concatenation

eviljeff reviewed (approved):
From a spot check, found a few places where `'abc' 'def'` could be collapsed into a single string.  I wouldn't bother fixing and making a new commit+ci+etc, just noting it down while I saw it.
r+ 🚢

willdurand commented:
I'll fix the issues you've found in a follow-up, thanks

willdurand commented:
> From a spot check, found a few places where `'abc' 'def'` could be collapsed into a single string. I wouldn't bother fixing and making a new commit+ci+etc, just noting it down while I saw it.

I filed https://github.com/mozilla/addons-server/issues/16139
